### PR TITLE
API: Fix missing keys in listZonesMetrics response

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR fixes https://github.com/apache/cloudstack/issues/9379
Fix missing keys in `listZonesMetrics`.

We use `BeanUtils.copyProperties` to copy the properties of `ZoneResponse` to `ZoneMetricsResponse`. Fields with missing getters are skipped from the response because of this.

https://github.com/apache/cloudstack/blob/56c661c1dff741ee9f7d741a7751f96504a16388/plugins/metrics/src/main/java/org/apache/cloudstack/metrics/MetricsServiceImpl.java#L930

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

File: api/src/main/java/org/apache/cloudstack/api/response/ZoneResponse.java [hunk 1]
```diff
@@ -344,6 +336,30 @@ public Map<String, String> getResourceDetails() {
...
+
+    public Integer getRouterPrivateInterfaceMaxMtu() {
+        return routerPrivateInterfaceMaxMtu;
+    }
+
+    public Integer getRouterPublicInterfaceMaxMtu() {
+        return routerPublicInterfaceMaxMtu;
+    }
+
+    public boolean getNsxEnabled() {
```
Comment (DaanHoogland) [inline api/src/main/java/org/apache/cloudstack/api/response/ZoneResponse.java:359]: it is an eternal discussion `is...()` vs. `get..()`, but as this is not a wrapper, isn't `is()` more appropriate?

Comment (vishesh92) [reply api/src/main/java/org/apache/cloudstack/api/response/ZoneResponse.java:359]: `BeanUtils.copyProperties` expects the names to be like `get...`. If I use `is...()`, it doesn't work.

Comment (vishesh92) [reply api/src/main/java/org/apache/cloudstack/api/response/ZoneResponse.java:359]: This happened for `localStorageEnabled` & `securityGroupsEnabled` which are `Boolean`. Not sure about `nsxEnabled` which is a primitive type (`boolean`).

**Comment by rohityadavcloud** [comment]
@vishesh92 LGTM didn't test it, can you raise this for 4.19 branch?

File: api/src/main/java/org/apache/cloudstack/api/response/ZoneResponse.java (hunk 1 above)
Comment (DaanHoogland) [reply api/src/main/java/org/apache/cloudstack/api/response/ZoneResponse.java]: as discussed off-line; let's use as many primitive as possible and expose those with `is..()` calls. Wrappers are useful only on optional output.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by vishesh92** [comment]
@blueorangutan package

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@vishesh92 also worth checking for list APIs & list metrics APIs for cluster and host? (and may VM and volumes)?

@blueorangutan test

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test rocky8 kvm-rocky8

**Review by rohityadavcloud — APPROVED** [review]
LGTM, the changes are only (a) getters, (b) typo fixes and (c) Boolean -> boolean

**Pull request merged**